src/python/turicreate/toolkits/sound_classifier/_tf_sound_classifier.py

```diff
@@ -25,67 +25,70 @@ def __init__(self, num_inputs, num_classes, custom_layer_sizes):
 
         """
 
-        _tf.reset_default_graph()
+        self.sc_graph = _tf.Graph()
 
-        self.num_classes = num_classes
+        with self.sc_graph.as_default():
 
-        self.x = _tf.placeholder("float", [None, 12288])
-        self.y = _tf.placeholder("float", [None, self.num_classes])
+            self.num_classes = num_classes
 
-        #xavier initialization
-        initializer = _tf.keras.initializers.glorot_uniform()
+            self.x = _tf.placeholder("float", [None, 12288])
+            self.y = _tf.placeholder("float", [None, self.num_classes])
 
-        weights = {}
-        biases = {}
-        self.names_of_layers = []
+            # Xavier initialization
+            initializer = _tf.keras.initializers.glorot_uniform()
 
-        # Create variables for customized layers
-        for i, cur_layer_size in enumerate(custom_layer_sizes):
+            weights = {}
+            biases = {}
+            self.names_of_layers = []
+
+            # Create variables for customized layers
+            for i, cur_layer_size in enumerate(custom_layer_sizes):
+                weight_name = 'sound_dense{}_weight'.format(i)
+                bias_name = 'sound_dense{}_bias'.format(i)
+                self.names_of_layers.append('dense{}'.format(i))
+                out_units = cur_layer_size
+                if i==0:
+                    in_units = num_inputs
+                weights[weight_name] = _tf.Variable(initializer([in_units, out_units]), name=weight_name)
+                biases[bias_name] = _tf.Variable(initializer([out_units]), name=bias_name)
+                in_units = out_units
+
+            i+=1
             weight_name = 'sound_dense{}_weight'.format(i)
             bias_name = 'sound_dense{}_bias'.format(i)
             self.names_of_layers.append('dense{}'.format(i))
```

Comment:
Per offline discussion, let's try a gentle refactor to avoid changing the indent level. But shouldn't these lines remain within the for loop?

Reply:
Working on indentation. These lines add the last layer so they are outside the for-loop. Strangely if I append `num_classes` to `custom_layer_sizes` to allow everything within the for-loop, things break. I'll try to reiterate over that again.

Reply:
Oh, I got fooled in the same way that the diff got fooled, due to the indentation...

Reply:
Yeah indentation makes diff look confusing